code/player/Player.cs

```diff
@@ -52,6 +66,8 @@ public override void Respawn()
             Role = RoleType.None;
             Credits = 0;
 
+            RemovePlayerCorpse();
+            TTTReborn.Gamemode.Game.Instance?.Round?.OnPlayerSpawn(this);
```

Comment:
How about using the EventSystem for things like this? <https://github.com/TTTReborn/ttt-reborn/commit/1776803a4b26d6614eba13b363bbc8a4a4c14a2e>

Reply:
Do you mean remove `Respawn()` and then move all logic to `public void OnPlayerSpawned()`?

Reply:
More or less something like this <https://github.com/TTTReborn/ttt-reborn/commit/1776803a4b26d6614eba13b363bbc8a4a4c14a2e#diff-d451f87d88459b7f181b1aa4bbd7846a4202c5650bd699912b88ff2906cacf37R30>

So you don't have to call it for every upcoming system individually, but once with the option of hooking functions easily up.

Reply:
We should really get that branched merged in. I added a TODO because the conflict resolution will be painful with the new files it adds.